fortitude/src/options.rs

```diff
@@ -0,0 +1,287 @@
+            # Enumerate all fixed violations.
+            show-fixes = true
+        "#
+    )]
+    pub show_fixes: Option<bool>,
+
+    /// Like [`fix`](#fix), but disables reporting on leftover violation. Implies [`fix`](#fix).
+    #[option(default = "false", value_type = "bool", example = "fix-only = true")]
+    pub fix_only: Option<bool>,
+
+    /// The style in which violation messages should be formatted: `"full"` (default)
+    /// (shows source), `"concise"`, `"grouped"` (group messages by file), `"json"`
+    /// (machine-readable), `"junit"` (machine-readable XML), `"github"` (GitHub
+    /// Actions annotations), `"gitlab"` (GitLab CI code quality report),
+    /// `"pylint"` (Pylint text format) or `"azure"` (Azure Pipeline logging commands).
+    #[option(
+        default = r#""full""#,
+        value_type = r#""full" | "concise" | "grouped" | "json" | "junit" | "github" | "gitlab" | "pylint" | "azure""#,
+        example = r#"
+            # Group violations by containing file.
+            output-format = "grouped"
+        "#
+    )]
+    pub output_format: Option<OutputFormat>,
+
+    /// Whether to enable preview mode. When preview mode is enabled, Fortitude will
+    /// use unstable rules, fixes, and formatting.
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            # Enable preview features.
+            preview = true
+        "#
+    )]
+    pub preview: Option<bool>,
+
+    /// Progress bar settings.
+    /// Options are "off" (default), "ascii", and "fancy"
+    #[option(
+        default = "off",
+        value_type = "str",
+        scope = "progress-bar",
+        example = r#"
+          # Enable unicode progress bar
+          progress-bar = "fancy"
+       "#
+    )]
+    pub progress_bar: Option<ProgressBar>,
+
+    // Rule selection
+    /// A list of rule codes or prefixes to ignore. Prefixes can specify exact
+    /// rules (like `T003` or `superfluous-implicit-none`), entire categories
+    /// (like `T` or `typing`), or anything in between.
+    ///
+    /// When breaking ties between enabled and disabled rules (via `select` and
+    /// `ignore`, respectively), more specific prefixes override less
+    /// specific prefixes.
+    #[option(
+        default = "[]",
+        value_type = "list[RuleSelector]",
+        example = r#"ignore = ["superfluous-implicit-none"]"#
+    )]
+    pub ignore: Option<Vec<RuleSelector>>,
+
+    // TODO: fix default and example when default rules decided!
+    /// A list of rule codes or prefixes to enable. Prefixes can specify exact
+    /// rules (like `T003` or `superfluous-implicit-none`), entire categories
+    /// (like `T` or `typing`), or anything in between.
+    ///
+    /// When breaking ties between enabled and disabled rules (via `select` and
+    /// `ignore`, respectively), more specific prefixes override less
+    /// specific prefixes.
+    #[option(
+        default = r#"["E4", "E7", "E9", "F"]"#,
```

Comment:
I see a TODO comment here to address that this needs to be updated, but I would have thought this would break things.

Reply:
Resolved in another comment...